rc/filetype/kotlin.kak

```diff
@@ -0,0 +1,202 @@
+# Oracle 2020, Java Platform, Standard Edition & Java Development Kit, Version 14 API Specification, viewed 8 September 2020, https://docs.oracle.com/en/java/javase/14/docs/api/index.html
+# ------------------------------------------------------------------------------------------------- #
+# ------------------------------------------------------------------------------------------------- #
+# File types
+# ‾‾‾‾‾‾‾‾‾‾
+hook global BufCreate .*[.](kt|kts)  %{
+  set-option buffer filetype kotlin
+}
+
+# ------------------------------------------------------------------------------------------------- #
+# Initialization
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+hook global WinSetOption filetype=kotlin %{
+  require-module kotlin
+
+  set-option -add window static_words %opt{main_static_words}
+  set-option -add window static_words %opt{docs_static_words}
+  set-option -add window static_words %opt{error_static_words}
+  set-option -add window static_words %opt{exception_static_words}
+
+  # cleanup trailing whitespaces when exiting insert mode
+  hook window ModeChange pop:insert:.* -group kotlin-trim-indent %{ try %{ execute-keys -draft <a-x>s^\h+$<ret>d } }
+  hook window InsertChar \n -group kotlin-indent kotlin-indent-on-new-line
+  hook window InsertChar \{ -group kotlin-indent kotlin-indent-on-opening-curly-brace
+  hook window InsertChar \} -group kotlin-indent kotlin-indent-on-closing-curly-brace
+
+  hook -once -always window WinSetOption filetype=.* %{ remove-hooks window kotlin-.+ }
+}
+
+hook -group kotlin-highlighter global WinSetOption filetype=kotlin %{
+  add-highlighter window/kotlin ref kotlin
+  add-highlighter window/ktor ref ktor
+  add-highlighter window/kdoc ref kdoc
+
+  hook -once -always window WinSetOption filetype=.* %{
+    remove-highlighter window/kotlin
+    remove-highlighter window/ktor
+    remove-highlighter window/kdoc
+  }
+}
+
+hook global BufSetOption filetype=kotlin %{
+  require-module kotlin
+
+  set-option buffer comment_line '//'
+  set-option buffer comment_block_begin '/*'
+  set-option buffer comment_block_end '*/'
+
+  hook -once -always buffer BufSetOption filetype=.* %{ remove-hooks buffer kotlin-.+ }
+}
+
+# ------------------------------------------------------------------------------------------------- #
+# Module
+# ‾‾‾‾‾‾
+provide-module kotlin %§
+# ------------------------------------------------------------------------------------------------- #
+add-highlighter shared/kotlin regions
+add-highlighter shared/kotlin/code default-region group
+add-highlighter shared/kotlin/string region %{(?<!')"} %{(?<!\\)(\\\\)*"} fill string
+add-highlighter shared/kotlin/comment region /\* \*/ fill comment
+add-highlighter shared/kotlin/inline_documentation region /// $ fill documentation
+add-highlighter shared/kotlin/line_comment region // $ fill comment
+
+add-highlighter shared/kotlin/code/kotlinAnnotations regex @\w+\b|\b\w+@(?=\{) 0:magenta
```

Comment:
All this paragraph is also using hard coded colors instead of faces, shoud be fixed.